src/ResourceManager/KeyVault/Commands.KeyVault/Commands/GetAzureKeyVaultCertificate.cs

```diff
@@ -146,5 +171,20 @@ private void GetAndWriteCertificatesVersions(string vaultName, string name, stri
                 WriteObject(pageResults, true);
             } while (!string.IsNullOrEmpty(options.NextLink));
         }
+
+        private void GetAndWriteDeletedCertificates( string vaultName )
+        {
+            KeyVaultObjectFilterOptions options = new KeyVaultObjectFilterOptions
+            {
+                VaultName = VaultName,
+                NextLink = null
+            };
+
+            do
+            {
+                var pageResults = DataServiceClient.GetDeletedCertificates(options);
```

Comment:
@dragav can we change this block to something like:

```cs
var pageResults = new List<DeletedCertificateIdentityItem>()
do
{
    pageResults.Add( DataServiceClient.GetDeletedCertificates( options ) );
} while ( !string.IsNullOrEmpty( options.NextLink ) );

WriteObject( pageResults, true );
```

this will ensure that all of the results are enumerated as a collection at once